garden_ai/app/garden.py

```diff
@@ -182,6 +143,74 @@ def create(
     return
 
 
+@garden_app.command(no_args_is_help=True)
+def search(
+    title: Optional[str] = typer.Option(
+        None, "-t", "--title", help="Title of a Garden"
+    ),
+    authors: Optional[List[str]] = typer.Option(
+        None, "-a", "--author", help="an author of the Garden"
+    ),
+    year: Optional[str] = typer.Option(
+        None, "-y", "--year", help="year the Garden was published"
+    ),
+    contributors: List[str] = typer.Option(
+        None,
+        "-c",
+        "--contributor",
+        help="a contributor to the Garden",
+    ),
+    description: Optional[str] = typer.Option(
+        None,
+        "-d",
+        "--description",
+        help="text in the description of the Garden you are searching for",
+    ),
+    tags: List[str] = typer.Option(
+        None,
+        "--tag",
+        help="A tag of the Garden",
+    ),
+    verbose: bool = typer.Option(
+        False, help="If true, print the query being passed to Globus Search."
+    ),
+    raw_query: Optional[str] = typer.Option(
+        None,
+        help=(
+            "Form your own Globus Search query directly. It will be passed to Search in advanced mode."
+            "See https://docs.globus.org/api/search/reference/get_query for more details."
+        ),
+    ),
+):
+    """Queries the Garden search index and prints matching results. All query components are ANDed together.
```

Comment:
Really like this help text, works as CLI help but even as a less structured docstring it clarifies what the actual code is doing too